sklearn/utils/estimator_checks.py

```diff
@@ -2052,8 +2054,25 @@ def check_estimators_data_not_an_array(name, estimator_orig, X, y):
     set_random_state(estimator_1)
     set_random_state(estimator_2)
 
-    y_ = NotAnArray(np.asarray(y))
-    X_ = NotAnArray(np.asarray(X))
+    if obj_type not in ["NotAnArray", 'PandasDataframe']:
+        raise ValueError("Data type {0} not supported".format(obj_type))
+
+    if obj_type == "NotAnArray":
+        y_ = NotAnArray(np.asarray(y))
+        X_ = NotAnArray(np.asarray(X))
+    else:
+        try:
+            import pandas as pd
```

Comment:
I did not use pytest.importorskip here because this is a utility file and the pytest module was not imported. I used the try/except block instead to be consistent with the check_sample_weights_pandas_series() function within the same file.

Reply:
Your concern is to avoid a dependency on pytest? I think that this is reasonable. Maybe a local import (inside the function).

Reply:
I tried to use `pytest.importorskip`, but it does not work in this case because, the estimator checks are not only used in tests, but also in other cases (e.g. doctest). Using `pytest.importorskip` would not avoid the ImportError in those cases.